src/review-todo.js

```diff
@@ -0,0 +1,212 @@
+let isDate = require("date-fns/isDate");
+
+function todo(todoName = "New Todo") {
```

Comment:
Done!

When I wrote this I had recently read [an article](https://tsherif.wordpress.com/2013/08/04/constructors-are-bad-for-javascript/) which was arguing for factory functions over constructor functions. I used factory functions with the idea from that article that it was more true to how the code actually worked, but I find myself bouncing between factory functions, constructors, and classes.

Do you have any tips or heuristics for picking which patterns to use when learning a language?

Reply:
I'd say that article lays out a reasonable cause to use factory functions instead. I wouldn't call either right or wrong here.

> Do you have any tips or heuristics for picking which patterns to use when learning a language?

This isn't limited to while you're learning 😄. It's a great question and doesn't have a satisfying answer. When learning, follow your guide/tutorial and try to understand the problem alternatives are trying to solve. Factory functions are an alternative to constructors because programmers might forget the `new` keyword and `instanceof` breaks if you modify the prototype outside the constructor. 

When you're more experienced your process might look like mine, which is a combination of
1. what I feel makes the most sense
1. what people around me are doing
1. and which pattern has tradeoffs that make the most sense in my use case. 

The easiest way to answer the first is to try different things and see what's clicks and where I make the most unforced errors and bugs. If you're diligent about `new` and not explicitly checking instances types, I'd say constructors are fine. (If you use polymorphism correctly, you shouldn't have to check types often).  The second is based  on culture of folks I'm working with. The third is the hardest one. Varies by project, familiarity, and clarity.